### ISSUE TITLE
Bug: prefer-set-has false positive for strings in JS

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [sindresorhus/eslint-plugin-unicorn#2216]
`prefer-set-has` seems to be giving false positives in cases where a string variable has an `.includes()` method called twice:

```js
const text = "abc".slice();
//    ~~~~
// `text` should be a `Set`, and use `text.has()` to check existence or non-existence.

text.includes("a") || text.includes("b");
```

Seen in real world code: [eslint/eslint@`59845bd` (#17515)](https://github.com/eslint/eslint/pull/17515/commits/59845bdb89eaf23ec028b2309fb6ef1dc1d895da#diff-c4eca4aa73ca33a81fc6bfbeb64ebfd27e369dfbeea7e36a73fff08daa46c1b9R252-R253)
